vxfreeswitch/voice.py

```diff
@@ -118,13 +120,32 @@ def stream_text_as_speech(self, message):
     def get_address(self):
         return self.uniquecallid
 
-    def output_message(self, text):
-        self.log("Sending text as speech: %r" % (text,))
-        return self.stream_text_as_speech(text)
-
-    def output_stream(self, url):
-        self.log("Playing back URL: %r" % (url,))
-        return self.playback(url)
+    def output_message(self, text, settings={}):
+        return self.stream_text_as_speech(text, settings=settings)
+
+    def output_stream(self, message, settings={}):
+        self.log("Playing back: %r" % (message,))
+        if settings.get('barge_in'):
+            terminator = settings.get('wait_for')
+            if settings.get('wait_for') is None:
+                # We just want to get 1 digit
+                minimum, maximum = 1, 1
+                terminator = "' '"
+            else:
+                # 128 is the maximum amount of digits that freeswitch can
+                # collect
+                minimum, maximum = 0, 128
+            tries = settings.get('tries', 1)
+            timeout = settings.get('time_gap', 3000)
+            # We have to have an invalid response message, so we set it to
+            # 1ms of silence
+            invalid_message = 'silence_stream://1'
+            return self.execute('play_and_get_digits', ' '.join([
+                str(minimum), str(maximum), str(tries), str(timeout),
+                str(terminator), message, invalid_message, self.VAR_NAME,
```

Comment:
This was also back when I got the result from the execute complete instead of through the DTMF things that we already have. This is where freeswitch stores the result when `play_and_get_digits` completes.